[ADD] add modules to use product packaging

laetitia-gangloff opened the pull request:
packaging_extended: to use uom in packaging
purchase_packaging: to improve supplierinfo with purchase uom and packaging and use them in purchase order
sale_packaging: to use packaging uom in sale order line

Commit:
[ADD] add modules to use product packaging

packaging_extended: to use uom in packaging
purchase_packaging: to improve supplierinfo with purchase uom and packaging and use them in purchase order
sale_packaging: to use packaging uom in sale order line

chrisdec commented:
Functional tests : 

It works very well regarding configuration and creation of manual purchase orders.

But I noticed a problem with the scheduler. The procurement generated by the scheduler (based on reordering rules) is correct but the linked PO is not. The supplier is correct but the packaging is not taken into account and thus unit of measures in the PO are not correct.

Commit:
[FIX] purchase_packaging : po from procurement

When Purchase Order are created from procurement take account of seller configuration.

laetitia-gangloff commented:
@chrisdec I make some corrections following your test.

nhomar commented on packaging_extended/__init__.py:
```diff
@@ -0,0 +1,5 @@
+
+from . import (
+    models,
+)
```
one line pliz.

nhomar commented:
Hello,

Can you change the name of packaging_extended I think name packaging_uom fit better to what it is doing (you can use the one you decide is just an example.

The words *extended *extra *plus do not say too much about what a module does.

Regards.

nhomar commented on purchase_packaging/__init__.py:
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
+
+from . import (
+    models,
```
one line pliz.

nhomar commented:
I do not like too much the rule, but there it is, can you split the PR is a PR per module pliz.

You can follow this steps.:

https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA

laetitia-gangloff commented:
@nhomar Thank you for your review ! I renamed packaging_extended in packaging_uom as you suggested.
I am not sure it is a good idea to split these 3 (little) modules in 3 PR.
Indeed, puchase_packaging and sale_packaging are dependent of packaging_uom, and can not be tested without it.
Is it possible to make a PR dependent of another PR to make test pass ?

pedrobaeza commented:
You can make copy branches that include the original commit of packaging_uom, and add commits for other module, and make the PR, so dependency is included. When the first PR will be merged, only rebasing you will get again a correct PR for the second and third module.

lmignon commented on purchase_packaging/tests/test_packaging.py:
```diff
@@ -0,0 +1,321 @@
+        self.assertAlmostEqual(sm.product_uom_qty, 16)
+
+    def test_procurement(self):
+        """ On supplierinfo set min_qty as 0
```
IMO it's better to describe the steps inside the code before each step

Commit:
[FIX] rename packaging_extended to packaging_uom

Commit:
[FIX] create purchase order from view

chrisdec commented:
:+1: (functional tests)

Commit:
[FIX] purchase_packaging : compute price_unit with the new uom / packaging_management : update readme

sbidoul commented:
@laetitia-gangloff @chrisdec can you improve the README's? The current proposed description explain the fields that are added, but for people not familiar with the intented use case, it's difficult to understand why the modules are useful.

A few sentences explaining what the module do that cannot be done in standard Odoo would help. And maybe a specific example.

Cedric-Pigeon commented on purchase_packaging/models/product.py:
```diff
@@ -0,0 +1,49 @@
+        return self.env.ref('product.product_uom_unit')
+
+    packaging_id = fields.Many2one('product.packaging', 'Logisitical Units')
+    product_uom = fields.Many2one(compute='_compute_product_uom',
```
better to call it product_uom_id, isn't it ?

laetitia-gangloff replied:
It is an override of the existing field, so I cannot rename it.

gurneyalex commented:
can you please rebase your PR? The travis build of 8.0 has recently been fixed.

laetitia-gangloff commented:
It is rebased. Thank you.

Commit:
When create po line set product_qty to avoid to reset qty to 1

Commit:
remove useless xml view

Commit:
correct product creation with packaging

Commit:
correct xml view

Commit:
Improve create of purchase order line to set it compatible with test_data

gurneyalex commented on purchase_packaging/models/purchase.py:
```diff
@@ -0,0 +1,238 @@
+        return vals
+
+    @api.model
+    def create(self, vals):
```
you need to add `@api.return('self', lambda rec: rec.id)`

gurneyalex commented on sale_packaging/models/sale_order_line.py:
```diff
@@ -0,0 +1,90 @@
+        return vals
+
+    @api.model
+    def create(self, vals):
```
you need to add @api.return here too

gurneyalex commented:
@laetitia-gangloff sorry for taking so long to review. A couple of missing decorators and this will be ok for me.

Commit:
[FIX] add missing decorators

laetitia-gangloff commented:
@gurneyalex Thank you very much for your review !
I corrected missing decorators.
I need to improve documentation ( :( ) for example Purchase packaging can be used to order by multiple.

And I hesitate to set product_qty on purchase order editable, and recompute the value according to minimal uom... Maybe later.

Commit:
purchase_packaging : fix supplier info view for create

lmignon commented:
@laetitia-gangloff IMO, what you propose should be done in an other PR. :+1: (Code + Functional)

Commit:
purchase_packaging : add post_init_hook to correctly initialize product_purchase_qty

gurneyalex commented:
:+1:

rousseldenis commented:
:+1: @gurneyalex Seems to be good for a merge ?
Plan to migrate in v10

lmignon commented:
I'll start the migration to v10

Commit:
[ADD] add setup for packaging_uom, purchase_packaging, sale_packaging

gurneyalex commented:
it's weird that travis is red... @Cedric-Pigeon could you try rebasing (again)?

rousseldenis commented:
@gurneyalex This is due to stock_change_qty_reason module error.

rousseldenis commented:
I will rebase it before merging

rousseldenis commented:
Closing this
